### PR TITLE
CAL-1339 - Data Explorer fix and Pagination

### PR DESCRIPTION
**Pull request by kyle-norris**

### CODE REVIEW

File: R/vars_manager.R [hunk 1]
```diff
@@ -139,91 +158,108 @@ get_matrix_var <- function(
...
+        i <- 1
+        for (col_name in 1:ncol(obj_pre)) {
+            col_names[i] <- as.character(col_name)
+            i <- i + 1
+        }
+        row_names <- list()
+        i <- 1
+        for (row_name in 1:nrow(obj_pre)) {
+            row_names[i] <- as.character(row_name)
+            i <- i + 1
```
Comment (pyb4430) [inline R/vars_manager.R:246]: can this be combined with the for loop on line 233?

File: R/vars_manager.R [hunk 2]
```diff
@@ -44,93 +44,112 @@ make_multi_dict <- function(
...
-            obj_sorted <- obj_filtered[order(tolower(obj_filtered), decreasing=decreasing, na.last=decreasing)]
-        } else {
-            obj_sorted <- obj_filtered[order(obj_filtered, decreasing=decreasing, na.last=decreasing)]
+        data <- list()
+        row_names <- list()
+
+        for (i in 1:length(obj_pre)) {
+            data[[i]] = list(as.character(obj_pre[i]))
+            if (is.null(names(obj_pre))) {
+                row_names[i] <- as.character(i)
```
Comment (pyb4430) [inline R/vars_manager.R:129]: I don't think this is quite right, `row_names` should match the page you're delivering (`start` to `end`)

File: R/vars_manager.R (hunk 1 above)
Comment (pyb4430) [reply R/vars_manager.R:246]: same concern as above, this doesn't account for the page offset, the row_names should match the page you're on

File: R/vars_manager.R [hunk 3]
```diff
@@ -243,151 +281,176 @@ get_dataframe_var <- function(
...
-        for (row_name in `if`(length(rownames(obj_pre)) > 1, rownames(obj_pre), 1:nrow(obj_pre))) {
-            row_names[i] <- as.character(row_name)
-            i <- i + 1
+        row_names <- list()
+        if (nrow(obj_pre) > 0) {
+            i <- 1
+            for (row_name in `if`(length(rownames(obj_pre)) > 1, rownames(obj_pre), 1:nrow(obj_pre))) {
+                row_names[i] <- as.character(row_name)
+                i <- i + 1
+            }
```
Comment (pyb4430) [inline R/vars_manager.R:423]: same as above, if there's no specific rownames then the row_names should reflect the page you're on

**Review by pyb4430 — CHANGES_REQUESTED** [review]
some row_names concerns, doesn't match callisto-python behavior

**Comment by kyle-norris** [comment]
you're definitely right about those row_names, they should be fixed now 👍

**Review by pyb4430 — APPROVED** [review]

**Pull request merged**